test/FastTests/Client/QueriesWithCustomFunctions.cs

```diff
@@ -2368,31 +2368,37 @@ select new
             using (var store = GetDocumentStore())
             {
                 await store.Operations.SendAsync(new PutCompareExchangeValueOperation<string>("Tom","Jerry", 0));
-                await store.Operations.SendAsync(new PutCompareExchangeValueOperation<string>("Zeus","Hera", 0));
-                
-                await store.Operations.SendAsync(new PutCompareExchangeValueOperation<string>("users/1","Thunderstorm", 0));
-                await store.Operations.SendAsync(new PutCompareExchangeValueOperation<string>("users/2","Cat", 0));
+                await store.Operations.SendAsync(new PutCompareExchangeValueOperation<string>("Hera","Zeus", 0));
                 
                 await store.Operations.SendAsync(new PutCompareExchangeValueOperation<string>("Jerry@gmail.com","users/2", 0));
                 await store.Operations.SendAsync(new PutCompareExchangeValueOperation<string>("Zeus@gmail.com","users/1", 0));
                             
+                await store.Operations.SendAsync(new PutCompareExchangeValueOperation<User>("ActiveUser",new User
+                {
+                    Name = "foo",
+                    IsActive = true
+                }, 0));
+                await store.Operations.SendAsync(new PutCompareExchangeValueOperation<User>("NotActiveUser",new User
+                {
+                    Name = "bar",
+                    IsActive = false
+                }, 0));
+                
                 using (var session = store.OpenSession())
                 {
-                    session.Store(new User { Name = "Jerry"}, "users/2");
-                    session.Store(new User { Name = "Zeus"}, "users/1");
+                    session.Store(new User { Name = "Jerry", IsActive = true}, "users/2");
+                    session.Store(new User { Name = "Zeus", IsActive = false}, "users/1");
                     session.SaveChanges();
                 }
 
                 using (var session = store.OpenSession())
                 {
-                    var rql = "FROM Users WHERE cmpxchg(Name) = \"Hera\"";
+                    var query = from u in session.Query<User>()
+                        where u.Name == RavenQuery.CmpXchg<string>("Hera")
```

Comment:
shouldn't this be handled like other methods? In LinqExtensions?

Reply:
Ok, this can be handled in LINQ like this, but for DocumentQuery we need to introduce IMO:
- `WhereCmpXchgEquals(fieldName, value)`
- `WhereCmpXchgNotEquals(fieldName, value)`
- `WhereCmpXchgGreaterThan(fieldName, value)`
- `WhereCmpXchgGreaterThanOrEqual(fieldName, value)`
- `WhereCmpXchgLessThan(fieldName, value)`
- `WhereCmpXchgLessThanOrEqual(fieldName, value)`

This will have to be implemented in `AbstractDocumentQuery` and will give you a better approach on handling this in LINQ, so you will not have to pass 'MethodCall' instance and will be able to handle this in a more straight-forward way.

Reply:
For now, we can just do Equals / not equals. We can support the rest after RTM.

Reply:
Probably a good idea would be to just add overload only to 
- `WhereEquals(fieldName, CmpXchg)`
- `WhereNotEquals(fieldName, CmpXchg)`
and introduce a class called `CmpXchg` that would also be used in LINQ queries:
```
from u in session.Query<User>()
where u.Name == CmpXchg.Create("Hera")
```
This would save a lot of troubles.